alwaysontop/main.js

```diff
@@ -62,13 +70,56 @@ function onAlwaysOnTopWindow(
 }
 
 /**
- * Returns the coordinates for the always on top window in order to display it
- * in the top right corner.
+ * Calculates the coordinates for a window (based on its current coordinates and
+ * size) to place it in the boundaries of a given screen.
+ *
+ * @param {Rectangle} windowRectangle - The current position and dimensions of
+ * the window.
+ * @param {Rectangle} screenRectangle - The position and dimensions of the
+ * screen.
+ * @returns {Point} - The new coordinates for the window.
+ */
+function positionWindowWithinScreenBoundaries(
+        windowRectangle,
+        screenRectangle) {
+    const maxY
+        = screenRectangle.y + screenRectangle.height - windowRectangle.height;
+    const maxX
+        = screenRectangle.x + screenRectangle.width - windowRectangle.width;
+
+    return {
+        x: Math.min(Math.max(windowRectangle.x, screenRectangle.x), maxX),
+        y: Math.min(Math.max(windowRectangle.y, screenRectangle.y), maxY)
+    };
```

Comment:
I think logic could use some comments because I'm having a hard time understanding with confidence what is happening. Could you also specify if a screens position origin is bottom-left/top-right/wherever?